cmd/ike/cmd/session_func_test.go

```diff
@@ -0,0 +1,96 @@
+package cmd_test
+
+import (
+	. "github.com/aslakknutsen/istio-workspace/cmd/ike/cmd"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var _ = Describe("Usage of session func", func() {
+
+	Context("checking required command arguments", func() {
+
+		var command *cobra.Command
+
+		BeforeEach(func() {
+			command = NewDevelopCmd()
+		})
+
+		It("should fail if namespace is not defined", func() {
+			_, err := ToOptions(removeFlagFromSet(command.Flags(), "namespace"))
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should fail if deployment is not defined", func() {
+			_, err := ToOptions(removeFlagFromSet(command.Flags(), "deployment"))
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should fail if session is not defined", func() {
+			_, err := ToOptions(removeFlagFromSet(command.Flags(), "session"))
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should fail if route is not defined", func() {
+			_, err := ToOptions(removeFlagFromSet(command.Flags(), "route"))
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should convert namespace if set", func() {
```

Comment:
How about moving it to other `Context`? Those are already testing conversion logic, not preconditions, as the current context they belong to ("checking required command arguments") might suggest.

Reply:
https://github.com/aslakknutsen/istio-workspace/pull/107